WordPress/gutenberg
# Site editor: style the selected template pattern

### What?

Fixes #65127

Enhance the template selector section by adding "selected" styling to active elements based on the issue raised in [GitHub Ticket #65127](https://github.com/WordPress/gutenberg/issues/65127).

### Why?

This improvement addresses the issue raised in [GitHub Ticket #65127]

Users had difficulty identifying which templates they had selected. Adding styling to active elements provides a clearer and better user experience.

### How?

To resolve this, two files were updated:

index.js [packages/block-editor/src/components/block-patterns-list/index.js]:
Modified the code to update the active pattern selection when a template is clicked. The is-active class is now added to active elements.

style.scss [packages/block-editor/src/components/block-patterns-list/style.scss ]: 
Styled the selected templates by targeting the is-active class, visually distinguishing them from unselected ones.

The proposed solution in the ticket could not resolve the issue using data-active-item because it wasn’t updating when templates were clicked. 
Hence, the JS code was updated to manage the active state and apply the is-active class on selected patterns. The styling in the CSS reflects these changes.

### Testing Instructions
1. Navigate to the template section in the site editor.
2. Change the template for any page.
3. Select a template pattern from the design section.
4. The selected pattern will have a blue border to indicate its active state.

### Testing Instructions for Keyboard

1. Navigate to the template section:
Use the Tab key to move through the site editor interface until you reach the template section.

2. Change the template:
Use the Tab key to navigate through the list of available templates.
Once focused on a template, press Enter to select it.

3. Select a template pattern:
Use Tab to navigate to the design section, where different template patterns are displayed.
Use the arrow keys (Up, Down, Left, Right) to move between the available patterns.
Press Enter to select a pattern. The active pattern should now be styled with a blue border.


![Screenshot](https://github.com/user-attachments/assets/849a4437-883b-4303-8a6e-d766b60a6272)
[](url)



Kindly let me know if anyone has any feedbacks or any improvements needed. 

Thanks,
Benazeer
